### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by KodiVerse**

### CODE REVIEW

File: test/util/rest/data/update.spec.js [hunk 1]
```diff
@@ -205,4 +246,29 @@ describe('Testing data formats', { useTmpDir: true }, () => {
...
+    await createAndPersistEntity(dir, {
+      id: offerId,
+      doc: index.data.remap('offer', { id: offerId, meta: { k1: 'v1' } })
+    });
+    instantiateIndex();
+    index.model.register('offer', updatedOfferModel);
+    index.index.register('offer', updatedOfferIndex);
+    await createAndPersistEntity(dir, {
+      id: offerId,
+      doc: index.data.remap('offer', { id: offerId, meta: { k1: 'v1' } })
```
Comment (KodiVerse) [inline test/util/rest/data/update.spec.js:252]: Meta does not show up in the document when I list it

File: test/util/rest/data/update.spec.js [hunk 2]
```diff
@@ -1,23 +1,54 @@
...
+      expect(index.index.versions.persist(dir)).to.equal(true);
+      expect(index.index.versions.load(dir)).to.equal(undefined);
+    };
+    const [offerModelPath, offerIndexPath] = ['models', 'indices']
+      .map((v) => path.join(__dirname, '..', '..', '..', `${v}`, 'offer.json'));
+    updatedOfferModel = sfs.smartRead(offerModelPath);
+    updatedOfferIndex = sfs.smartRead(offerIndexPath);
+    updatedOfferModel.fields.subhead = 'string';
+    updatedOfferIndex.fields.push('subhead');
+    getMetaFromDocs = async (idx) => {
```
Comment (simlu) [inline test/util/rest/data/update.spec.js:37]:
```suggestion
    queryAlias = async (idx) => {
```

File: test/util/rest/data/update.spec.js [hunk 3]
```diff
@@ -205,4 +236,33 @@ describe('Testing data formats', { useTmpDir: true }, () => {
       }
     });
   });
+
+  it('Testing update with signature in two versions', async ({ dir }) => {
+    expect(await index.rest.data.update('offer', [{
+      action: 'update',
+      doc: index.data.remap('offer', { id: offerId, meta: { k1: 'v1' } })
+    }])).to.equal(true);
+    expect(await index.rest.data.refresh('offer')).to.equal(true);
```
Comment (simlu) [inline test/util/rest/data/update.spec.js:245]:
```suggestion
```

File: test/util/rest/data/update.spec.js [hunk 4]
```diff
@@ -205,4 +236,33 @@ describe('Testing data formats', { useTmpDir: true }, () => {
       }
     });
   });
+
+  it('Testing update with signature in two versions', async ({ dir }) => {
+    expect(await index.rest.data.update('offer', [{
+      action: 'update',
+      doc: index.data.remap('offer', { id: offerId, meta: { k1: 'v1' } })
+    }])).to.equal(true);
```
Comment (simlu) [inline test/util/rest/data/update.spec.js:244]:
```suggestion
```

**Review by simlu — APPROVED** [review]

**Pull request merged**